### PR TITLE
Added support for request filtering

### PR DESCRIPTION
**Pull request by TobiasDeBruijn**
Changes:
- Add support for request Method filtering
- Rexports http::Method
- By default, all Methods are ratelimited, only when a user specifies one is the rest ignored --> No backwards compatibility issues

Solves #9

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ categories = ["web-programming::http-server"]
 
 [dependencies]
 actix-web = { version = "4.0.0-beta.9", default-features = false }
+http = "^0.2"
```
Comment (AaronErhardt) [inline Cargo.toml]: I don't know about the latest changes in the actix beta, but I think the re-export for `http` is still there so using `actix_web::http` would be better than introducing new deps.

File: src/lib.rs [hunk 2]
```diff
@@ -135,8 +141,9 @@ const DEFAULT_BURST_SIZE: u32 = 8;
 /// ```
 #[derive(Clone, PartialEq, Eq, Debug)]
 pub struct GovernorConfigBuilder {
-    period: Duration,
+    period:     Duration,
```
Comment (AaronErhardt) [inline src/lib.rs]: Indentation looks odd here. Maybe run `cargo fmt`?

File: src/tests.rs [hunk 3]
```diff
@@ -1,7 +1,8 @@
 use actix_web::{dev::Service, http::StatusCode, test, web, App, HttpResponse, Responder};
 
-#[test]
-fn builder_test() {
+// This won't compile without `async` for some reason
```
Comment (AaronErhardt) [inline src/tests.rs]: Interesting... this should not be async and I'm against changing this test unless there's a good reason.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks very well! There are just a few things I'd like to adjust.

File: src/tests.rs (hunk 3 above)
Comment (TobiasDeBruijn) [reply src/tests.rs]: I have no clue why it happens, but it wasn't compiling for me on 1.56.1, I'll do some research though.

File: Cargo.toml (hunk 1 above)
Comment (TobiasDeBruijn) [reply Cargo.toml]: Good catch, didn't know that was there!

File: src/lib.rs (hunk 2 above)
Comment (TobiasDeBruijn) [reply src/lib.rs]: Ah yeah that's my own doing. I tend to aline them all on 1 line. Will fix

File: src/tests.rs (hunk 3 above)
Comment (TobiasDeBruijn) [reply src/tests.rs]: Hm it seems this was an error on IntelliJ's part. The solution is to move the `use actix_web::test` into the function body. Something fishy is going on there.

I've included that in the commit so future users will not run into the issue

**Comment by AaronErhardt** [comment]
Thanks! I'll take care of the back port and create a new release once I'm back from work.

**Comment by TobiasDeBruijn** [comment]
Sounds good, thanks!

**Pull request merged**